src/components/CertificateTemplates/tlds/sg/edu/nus/NUSTS-GENERAL-2019/integration.spec.js

```diff
@@ -0,0 +1,41 @@
+import { Selector } from "testcafe";
+import { readFileSync } from "fs";
+import { join } from "path";
+import { getData } from "@govtechsg/open-attestation";
+
+fixture("National University of Singapore").page`http://localhost:3000`;
+
+const Certificate = "./sample.opencert";
+const RenderedCertificate = Selector("#rendered-certificate");
+
+const validateTextContent = async (t, component, texts) =>
+  texts.reduce(
+    async (prev, curr) => t.expect(component.textContent).contains(curr),
+    Promise.resolve()
+  );
+
+test("NUSTS-GENERAL-2019 transcript is rendered correctly", async t => {
+  // Inject javascript and execute window.opencerts.renderDocument
+  const certificateContent = getData(
+    JSON.parse(readFileSync(join(__dirname, Certificate)).toString())
+  );
+  await t.eval(() => window.opencerts.renderDocument(certificateContent), {
+    dependencies: { certificateContent }
+  });
+
+  // Check content of window.opencerts.templates
+  await t.wait(500);
+  const templates = await t.eval(() => window.opencerts.getTemplates());
+  await t
+    .expect(templates)
+    .eql([{ id: "transcript", label: "Transcript", template: undefined }]);
+
```

Comment:
you should consider testing more of the content on the page to validate that your formatting functions work correctly

Reply:
Done as recommended